odoo/tools/config.py

```diff
@@ -349,15 +344,51 @@ def __init__(self, fname=None):
                              type="int")
             parser.add_option_group(group)
 
-        # Copy all optparse options (i.e. MyOption) into self.options.
-        for group in parser.option_groups:
-            for option in group.option_list:
-                if option.dest not in self.options:
-                    self.options[option.dest] = option.my_default
-                    self.casts[option.dest] = option
-
-        # generate default config
-        self._parse_config()
+        # Config file only options
+        def add_file_only_option(**kwargs):
+            # they are discarded at runtime
+            # limitation of optparse: there must be at least one short/long opt.
+            opt = '--' + kwargs['dest'].replace('_', '-')
+            return parser.add_option(opt, **kwargs, cli_loadable=False, help=hidden)
+        add_file_only_option(dest='admin_passwd', my_default='admin')
+        add_file_only_option(dest='csv_internal_sep', my_default=',')
+        add_file_only_option(dest='publisher_warranty_url', my_default='http://services.openerp.com/publisher-warranty/', file_exportable=False)
+        add_file_only_option(dest='reportgz', action='store_true', my_default=False)
+        add_file_only_option(dest='websocket_keep_alive_timeout', type='int', my_default=3600)
+        add_file_only_option(dest='websocket_rate_limit_burst', type='int', my_default=10)
+        add_file_only_option(dest='websocket_rate_limit_delay', type='float', my_default=0.2)
+        del add_file_only_option
```

Comment:
```suggestion
        class FileOption(MyOption):
            def _check_opt_strings(self, opts):
                if opts:
                    raise TypeError("No option can be supplied")

            def _set_opt_strings(self, opts):
                return

        parser.add_option(FileOption(dest='admin_passwd', my_default='admin'))
        parser.add_option(FileOption(dest='csv_internal_sep', my_default=','))
        parser.add_option(FileOption(dest='publisher_warranty_url', my_default='http://services.openerp.com/publisher-warranty/', file_exportable=False))
        parser.add_option(FileOption(dest='reportgz', action='store_true', my_default=False))
        parser.add_option(FileOption(dest='websocket_keep_alive_timeout', type='int', my_default=3600))
        parser.add_option(FileOption(dest='websocket_rate_limit_burst', type='int', my_default=10))
        parser.add_option(FileOption(dest='websocket_rate_limit_delay', type='float', my_default=0.2))
```

Then you should be able to remove all the `cli_loadable` stuff as well has the `die(option_value is not None, f"invalid parameter` part. May be subjective but I prefer this solution. :) 

Note that you can del the FileOption afterward or define it next to "MyOption"
```
14:38:22  ~/osrc/master/odoo  master-config-cleanup-juc ≠ master 130 odoo -d test_config  --admin-passwd 123
Usage: odoo-bin server [options]

odoo-bin server: error: no such option: --admin-passwd
```